apps/android/app/src/main/java/chat/simplex/app/SimplexService.kt

```diff
@@ -62,7 +81,7 @@ class SimplexService: Service() {
 
   private fun startService() {
     Log.d(TAG, "SimplexService startService")
-    if (isServiceStarted || isStartingService) return
+    if (wakeLock != null || isStartingService) return
```

Comment:
I'm trying to understand what this achieves - is the `wakeLock != null` check added as a replacement for `isServiceStarted` or was it required before as well but missed? Is this line a sanity check or more of a fix?

Reply:
The behaviour of `isServiceStarted` variable is changed. Before that it was set `true` inside `startService` and that's why it was ok to check `if (isServiceStarted...` but now I need `isServiceStarted` following different logic: it has to be set `true` once the service is fully started and it has to be locked inside `companion object`. It is needed because I stop the service from that object to prevent unwanted behaviour caused by `stopSelf` function in which case it's called before calling `startForeground`.

So `wakeLock != null` is just a replacement of previously working logic that used `isServiceStarted` check because `wakeLock` will not be null only when it's set inside `startService` (it's set almost at the same place as `isServiceStarted` before).